core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -47,7 +47,8 @@ private[v1] class AllStagesResource(ui: SparkUI) {
         listener.stageIdToData.get((stageInfo.stageId, stageInfo.attemptId))
       }
     } yield {
-      AllStagesResource.stageUiToStageData(status, stageInfo, stageUiData, includeDetails = false)
+      AllStagesResource.stageUiToStageData(
+        status, stageInfo, stageUiData, includeDetails = false, Some(ui))
```

Comment:
It's not a good idea to pass in `SparkUI` to only get `lastUpdate`, the API looks weird to add this `SparkUI` argument, the fix here only just make it work. It is better to add one more field in `StageUIData` or `TaskUIData` if possible.